### PR TITLE
run: do not block on greenlets after command exits

### PR DESCRIPTION
**Pull request by batrick**
The stdout/stderr greenlets will not necessarily exit when the command does if
child processes are stuck in an uninterruptible sleep. For example, the fsx.sh
workunit spawns fsx processes that may be left behind in the D state after
/bin/timeout kills fsx.sh. These are connected to the stdout/stderr pipes which
prevent the greenlets from exiting normally.

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

**Comment by batrick** [comment]
Example: /ceph/teuthology-archive/pdonnell-2018-10-01_03:19:12-multimds-wip-pdonnell-testing-20181001.011252-distro-basic-smithi/3090388/teuthology.log

**Comment by batrick** [comment]
I don't see why the build is broken or how it could be caused by this PR.

File: teuthology/orchestra/run.py [hunk 1]
```diff
@@ -139,10 +139,16 @@ def wait(self):
...
         """
-        for greenlet in self.greenlets:
-            greenlet.get()
 
         status = self._get_exitstatus()
+        if status != 0:
+            log.debug("got remote process result: {}".format(status))
+        for greenlet in self.greenlets:
+            try:
+                s = greenlet.get(block=True,timeout=60)
```
Comment (tchaikov) [inline teuthology/orchestra/run.py]:
```
teuthology/orchestra/run.py:148:17: F841 local variable 's' is assigned to but never used
```

Comment (batrick) [reply teuthology/orchestra/run.py]: Ah, thanks @tchaikov

**Comment by vasukulkarni** [comment]
I think its a flake8 issue,  line 148: s initialized but never used

**Pull request merged**